### PR TITLE
add controller reachable to the vault lock timeout calculation

### PR DESCRIPTION
**Pull request by naiming-zededa**
Signed-off-by: Naiming Shen <naiming@zededa.com>

### CODE REVIEW

**Review by eriknordmark — COMMENTED** [review]
Don't you need to remove the code which currently sets vaultTestStartTime?

**Comment by naiming-zededa** [comment]
> Don't you need to remove the code which currently sets vaultTestStartTime?

I decided not to. currently when the first time got status from vaultmgr, we set this. if i remove this, and we got to controller first, but still not get from vaultmgr yet (unlikely, but...). So the current patch is still to keep this, and if this is non-zero, and we get to cloud, then we reset this starttime count; but if we get to cloud first, then we still wait for this old code to set this time. for the check unlock timer function, both needs to be valid in order to count the time lapse of 300 seconds

File: pkg/pillar/cmd/nodeagent/handletimers.go [hunk 1]
```diff
@@ -116,7 +116,10 @@ func handleRebootOnVaultLocked(ctxPtr *nodeagentContext) {
 	//This could be due to remote attestation taking time
 	//Check if we have crossed cut off time
 	vaultCutOffTime := ctxPtr.globalConfig.GlobalValueInt(types.VaultReadyCutOffTime)
-	timePassed := ctxPtr.timeTickCount - ctxPtr.vaultTestStartTime
+	var timePassed uint32
+	if ctxPtr.vaultTestStartTime > 0 && ctxPtr.configGetSuccess {
```
Comment (eriknordmark) [inline pkg/pillar/cmd/nodeagent/handletimers.go:120]: How about at least comment saying "check vaultTestStartTime to make sure we got a notification from vaultmgr"?

But would be easier to understand if we had a ctxPtr.vaultmgrReported to parallel ctxPtr.configGetSuccess.

**Comment by eriknordmark** [comment]
> > Don't you need to remove the code which currently sets vaultTestStartTime?
> 
> I decided not to. currently when the first time got status from vaultmgr, we set this. if i remove this, and we got to controller first, but still not get from vaultmgr yet (unlikely, but...). So the current patch is still to keep this, and if this is non-zero, and we get to cloud, then we reset this starttime count; but if we get to cloud first, then we still wait for this old code to set this time. for the check unlock timer function, both needs to be valid in order to count the time lapse of 300 seconds

Seems hard to understand that is what it does. See suggestion in comment on how to make this more clear and not rely on the value being > 0.

**Comment by naiming-zededa** [comment]
Ok. added the vaultmgrReported.

File: pkg/pillar/cmd/nodeagent/nodeagent.go [hunk 2]
```diff
@@ -655,6 +657,7 @@ func handleVaultStatusImpl(ctxArg interface{}, key string,
 	ctx := ctxArg.(*nodeagentContext)
 	vault := statusArg.(types.VaultStatus)
 
+	ctx.vaultmgrReported = true
 	if ctx.vaultTestStartTime == 0 {
 		//First update from vaultmgr, record it as test start time
```
Comment (eriknordmark) [inline pkg/pillar/cmd/nodeagent/nodeagent.go:662]: The more logical would be if !vaultMgrReported { vaultMgrReported = true; record test start time}

File: pkg/pillar/cmd/nodeagent/handletimers.go [hunk 3]
```diff
@@ -197,6 +200,11 @@ func updateZedagentCloudConnectStatus(ctxPtr *nodeagentContext,
 	case types.ConfigGetSuccess:
 		log.Functionf("Config get from controller, is successful")
 		ctxPtr.lastControllerReachableTime = ctxPtr.timeTickCount
+		if !ctxPtr.configGetSuccess && ctxPtr.vaultTestStartTime > 0 {
```
Comment (eriknordmark) [inline pkg/pillar/cmd/nodeagent/handletimers.go:203]: More logical thing would be to check vaultmgrreported instead of vaultTestStartTime.

**Review by eriknordmark — APPROVED** [review]
Let's run the tests in parallel with the conversation

File: pkg/pillar/cmd/nodeagent/handletimers.go (hunk 3 above)
Comment (naiming-zededa) [reply pkg/pillar/cmd/nodeagent/handletimers.go]: Sure. Changed

**Review by rvs — APPROVED** [review]
LGTM

**Pull request merged**